lib/aws/rails/sqs_active_job/executor.rb

```diff
@@ -16,13 +16,18 @@ class Executor
            fallback_policy: :caller_runs # slow down the producer thread
         }.freeze
 
-        def initialize(options = {})
+        def initialize(options = {}, refresh_timeout = nil)
           @executor = Concurrent::ThreadPoolExecutor.new(DEFAULTS.merge(options))
+          # Monitor threads used to refresh visiblity
+          @refresh_timeout = refresh_timeout
+          @monitor = Concurrent::ThreadPoolExecutor.new(DEFAULTS.merge(options)) if refresh_timeout
           @logger = options[:logger] || ActiveSupport::Logger.new(STDOUT)
         end
 
         # TODO: Consider catching the exception and sleeping instead of using :caller_runs
         def execute(message)
+          # Used to tell the visibilty refresh thread to give up
+          poison_pill = false
```

Comment:
The name `poison_pill` here is a little confusing  - and I'm not sure I understand the logic below in the `@monitor.post` block.  The monitor will update the visibility `while poison_pill`, but `poison_pill` starts as false and is only set to true once execution (successfully or unsuccessfully) is complete.  

Maybe consider updating this to be `job_executing` and set it to `true` initially, then `false` in the ensure block.

Reply:
My bad, flipped it around in my brain, due to bad naming 😆 Fixing.